pyproject.toml

```diff
@@ -38,76 +38,76 @@ priority = "supplemental"
 
 [tool.poetry.dependencies]
 python = ">=3.8.10, <3.13"
-aiofiles = "24.1.0"  # Async IO for files
-aiohttp = "3.9.5"  # HTTP server for full node rpc
-aiosqlite = "0.20.0"  # asyncio wrapper for sqlite, to store blocks
-anyio = "4.3.0"
-bitstring = "4.1.4"  # Binary data management library
-boto3 = "1.34.143"  # AWS S3 for Data Layer S3 plugin
-chiabip158 = "1.5.1"  # bip158-style wallet filters
-chiapos = "2.0.4"  # proof of space
-chia_rs = "0.13.0"
-chiavdf = "1.1.4"  # timelord and vdf verification
-click = "8.1.7"  # For the CLI
-clvm = "0.9.10"
-clvm_tools = "0.4.9"  # Currying Program.to other conveniences
-clvm_tools_rs = "0.1.43"  # Rust implementation of clvm_tools' compiler
-colorama = "0.4.6"  # Colorizes terminal output
-colorlog = "6.8.2"  # Adds color to logs
-concurrent_log_handler = "0.9.25"  # Concurrently log and rotate logs
-cryptography = "43.0.1"  # Python cryptography library for TLS - keyring conflict
-dnslib = "0.9.25"  # dns lib
-dnspython = "2.6.1"  # Query DNS seeds
-filelock = "3.15.4"  # For reading and writing config multiprocess and multithread safely  (non-reentrant locks)
-keyring = "25.2.1"  # Store keys in MacOS Keychain, Windows Credential Locker
-packaging = "24.0"
-pip = "24.2"
+aiofiles = ">=24.1.0"  # Async IO for files
```

Comment:
I personally started with `*` since we have no particular ties the the version we are exactly pinning now.  I switched to `>=` since Earle asked for it.  There's doubt in the Python community that upper version limits actually net a better ecosystem.  Yeah, SemVer, I'm all about it.  But, if you strictly follow it then you can end up with many major version bumps over minor corner case fixes that won't affect the vast majority of dependees. Then some of those dependees aren't really well maintained and have upper version limits.  Then nobody can install anything sensible anymore.  The flip side is more accidental breakage...  but.  Anyone can _add_ a constraint (an upper limit for a package) while nobody can take an existing constraint away from a package they don't own (like an upper limit).  And yes, there are issues talking about adding the ability to override other package's dependency constraints.  `:]`

So yes, we could `^` if we wanted to constrain the upper bound.  Given that we have a lock and require PRs, we'll be testing every update to the lock the same regardless of it being major, minor, or patch and also regardless of what constraint we use here.

There are cases, like a user needing to do a local version update for whatever reason, where better describing our limits of compatibility would be good.  Today, we provide no guidance on that since we pin version in both the lock and the project definition.  So, I'm not clear that just dropping the constraints in the project definition makes things particularly worse.